archunit/src/test/java/com/tngtech/archunit/core/domain/JavaClassTransitiveDependenciesTest.java

```diff
@@ -0,0 +1,148 @@
+
+        assertThatDependencies(classes.get(c).getTransitiveDependenciesFromSelfWithinPackage(PACKAGE))
+                .contain(c, Object.class)
+                .contain(c, d)
+                    .contain(d, Object.class)
+                    .contain(d, String.class)
+                .containOnlyTargetClasses(d, Object.class, String.class);
+    }
+
+
+    static class CyclicGraph {
+        static class A {
+            B b;
+            C[] c;
+            D d;
+        }
+
+        static class B {
+            Integer i;
+        }
+
+        static class C {
+            A a;
+        }
+
+        static class D {
+            E e;
+        }
+
+        static class E {
+            A a;
+            String s;
+        }
+    }
+
+    @Test
+    public void findsTransitiveDependenciesInCyclicGraph() {
+        Class<?> a = CyclicGraph.A.class;
+        Class<?> b = CyclicGraph.B.class;
+        Class<?> c = CyclicGraph.C.class;
+        Class<?> d = CyclicGraph.D.class;
+        Class<?> e = CyclicGraph.E.class;
+        JavaClasses classes = new ClassFileImporter().importClasses(a, b, c, d, e);
+        Class<?> cArray = CyclicGraph.C[].class;
+
+        assertThatDependencies(classes.get(a).getTransitiveDependenciesFromSelfWithinPackage(PACKAGE))
+                .contain(a, Object.class)
+                .contain(a, b)
+                    .contain(b, Object.class)
+                    .contain(b, Integer.class)
+                .contain(a, cArray)
+                    .contain(c, Object.class)
+                    .contain(c, a)
+                .contain(a, d)
+                    .contain(d, Object.class)
+                    .contain(d, e)
+                        .contain(e, Object.class)
+                        .contain(e, a)
+                        .contain(e, String.class)
+                .containOnlyTargetClasses(a, b, cArray, d, e, Object.class, Integer.class, String.class);
+
+        assertThatDependencies(classes.get(c).getTransitiveDependenciesFromSelfWithinPackage(PACKAGE))
+                .contain(c, Object.class)
+                .contain(c, a)
+                    .contain(a, Object.class)
+                    .contain(a, b)
+                        .contain(b, Object.class)
+                        .contain(b, Integer.class)
+                    .contain(a, cArray)
+                    .contain(a, d)
+                        .contain(d, Object.class)
+                        .contain(d, e)
+                            .contain(e, Object.class)
+                            .contain(e, a)
+                            .contain(e, String.class)
+                .containOnlyTargetClasses(a, b, cArray, d, e, Object.class, Integer.class, String.class);
+
+        assertThatDependencies(classes.get(d).getTransitiveDependenciesFromSelfWithinPackage(PACKAGE))
+                .contain(d, Object.class)
+                .contain(d, e)
+                    .contain(e, Object.class)
```

Comment:
I would imagine that one shortcut "format" would destroy all this nice indent? :wink: If yes, I think there are some "disable formatter" comments for IntelliJ. Just because I know that I repeatedly format everything like a robot, so it would be really annoying to me to always watch out for this place :wink:

Reply:
I've now added a `// @formatter:off`, but it's up to you whether your IDE respects that or not. 😉

Reply:
I like the formatting and my IDE respects it :wink: